src/osd/PG.cc

```diff
@@ -3725,8 +3727,7 @@ void PG::_scan_snaps(ScrubMap &smap)
 }
 
 /*
- * build a scrub map over a chunk without releasing the lock
- * only used by chunky scrub
+ * build a scrub map over a chunk
  */
```

Comment:
This method iirc is triggered by a OpRequestRef event in the OpWQ -- queued in handle_replica_op<MOSDRepScrub,...> called from OSD::dispatch_op_fast.  I think what you want to do is replace that call to the templated handle_replica_op with a new custom one which queues an event you can requeue until you've finished the scan.